tests/tests.rs

```diff
@@ -106,25 +112,35 @@ fn run_compile_test(
             Style::Type => {}
         }
     }
-    match language {
-        Language::Cxx => {
-            output.push(format!("{}.cpp", name));
-        }
+
+    let ext = match language {
+        Language::Cxx => "cpp",
         Language::C => {
-            output.push(format!("{}.c", name));
+            if cpp_compat {
+                "compat.c"
+            } else {
+                "c"
+            }
         }
-    }
+    };
+
+    output.push(format!("{}.{}", name, ext));
 
-    run_cbindgen(profile, path, &output, language, style);
+    run_cbindgen(profile, path, &output, language, cpp_compat, style);
     compile(&output, language);
+
+    if language == Language::C && cpp_compat {
+        compile(&output, Language::Cxx)
+    }
 }
 
 fn test_file(profile: &'static str, name: &'static str, filename: &'static str) {
     let test = Path::new(filename);
     for style in &[Style::Type, Style::Tag, Style::Both] {
-        run_compile_test(profile, name, &test, Language::C, Some(*style));
+        run_compile_test(profile, name, &test, Language::C, true, Some(*style));
```

Comment:
Maybe worth making the new bool argument an `enum`, so it's clearer what it means from here.

Otherwise at least `/* cpp_compat = */ true`?

Reply:
I decided to add a second for loop: `for cpp_compat in &[true, false]`, to make it clear that we are iterating over the `cpp_compat` option.

I thought that to be a better alternative than introducing an enum, as an enum would mean that the entire codebase becomes riddled with `if config.cpp_compat == Compat::On` statements, as opposed to a cleaner boolean check.

Reply:
Oh I totally mean to keep the enum scoped to the tests.rs file, so the only place that would have to do that is the place that adds the argument. But sure, separate loop looks good too